### PR TITLE
Remove the `DAILY` enum value from `SyncGroup`

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://app.clubhouse.io/simpledotorg/story/4148/remove-separation-between-daily-and-frequent-syncs

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/sync/DataSync.kt [hunk 1]
```diff
@@ -66,20 +66,18 @@ class DataSync(
...
 
-  private fun syncsForGroup(syncGroup: SyncGroup): Single<List<SyncResult>> {
-    val syncsInGroup = modelSyncs.filter { it.syncConfig().syncGroup == syncGroup }
-
+  private fun syncsForGroup(): Single<List<SyncResult>> {
     return Single
         .fromCallable { userSession.loggedInUserImmediate().toOptional() }
         .subscribeOn(schedulersProvider.io())
-        .compose(filterSyncsThatRequireAuthentication(syncsInGroup))
+        .compose(filterSyncsThatRequireAuthentication(modelSyncs))
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/sync/DataSync.kt:70]: the `filterSyncsThatRequireAuthentication` function parameter name is not changed in the body. Can you update?

File: app/src/test/java/org/simple/clinic/sync/DataSyncTest.kt [hunk 2]
```diff
@@ -38,26 +38,16 @@ class DataSyncTest {
...
-      syncGroup = SyncGroup.DAILY
-  )
-
   @Test
-  fun `when syncing everything, all the model syncs should be invoked`() {
+  fun `when syncing, all the model syncs should be invoked`() {
     val modelSync1 = FakeModelSync(
         _name = "sync1",
-        config = dailySyncConfig
+        config = frequentSyncConfig
```
Comment (Honey14) [inline app/src/test/java/org/simple/clinic/sync/DataSyncTest.kt:45]: So we'll be removing `SyncInterval` and `SyncGroup` from the FakeModelSync when we finish removing frequent type as well, right?

**Review by Honey14 — CHANGES_REQUESTED** [review]
Looks good overall. Requested one change. PTAL

File: app/src/test/java/org/simple/clinic/sync/DataSyncTest.kt (hunk 2 above)
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/sync/DataSyncTest.kt:45]: Yup, that is correct.

File: app/src/main/java/org/simple/clinic/sync/DataSync.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/sync/DataSync.kt:70]: Good catch, fixing this.

**Review by Honey14 — APPROVED** [review]
Looks good

**Pull request merged**